api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -114,6 +121,10 @@ message RateLimitResponse {
       // Point in time at which the quota expires.
       google.protobuf.Timestamp valid_until = 2;
     }
+
+    // The unique id that is associated with each Quota either at individual descriptor level or whole descriptor set level.
+    // This can be used to share the same quota among multiple descriptors or descriptor sets.
```

Comment:
I'm not sure I follow this? Where is ID used by the client? From the client perspective this seems like something for logging?

Reply:
It is not for logging. Some context here: Our control plane (for this feature) doesn't support CEL expression yet (please note this will be supported eventually and it is already on the related team's schedule), hence for policy with Disjunction : `match: "request.headers['environment'] == 'staging' || request.headers['environment'] == 'dev'"`, the [request_headers action](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-ratelimit-action-requestheaders) produces a distinct list of descriptors for each possible value of the ‘environment’ header. So, we add the unique id to correctly correlate the mapping between descriptor list and quota.  Use the match above as example: 

We have Descriptor list 1 = `[[(“environment”, “staging”)], [(“policy”: “default”)]` and Descriptor list 2 = `[[(“environment”, “environment”)], [(“policy”: “default”)]`. Without CEL expression , we cannot tell that they are using the same quota. So here we add the id filed and the RLS server will map those two D1 and D2 to the same quota using this `id`

Reply:
Sorry for being dense, but I don't understand what the client is supposed to do with ID. Can you clarify?

Reply:
It is sever return the `Quota` with `ID` information. As the client, Envoy see that the id is same across D1 and D2, i.e., same quota will be applied to D1 and D2, we will keep this information (like a mapping : Quota1 <-> {D1, D2}). in the future if either D1 or D2 request comes, we will just use this quota in the cache instead of querying the RLS sever. 
Does this make sense?

Reply:
What if the quota values are different? Are they refreshed?

Either way can you flesh this section out substantially to explain what the client is supposed to do?

Reply:
If the quota values are different, the RLS server should not return the same id. 

I added more comments about this section. PTAL. Thanks!